File: nj2k/testData/newJ2k/javaStreamsApi/kt-21467.java
// RUNTIME_WITH_FULL_JDK

import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.Stream;

class Test {
    public void main() {
        List<String> activities = Stream.of("12")
                .map(v -> v + "nya")
                .filter(v -> v != null)
                .flatMap(v -> Stream.of(v)
                        .flatMap(s -> Stream.of(s))
                ).filter(v -> {
                    String name = v.getClass().getName();
                    if (name.equals("name")) {
                        return false;
                    }
                    return !name.equals("other_name");
                })
                .collect(Collectors.toList());
    }
}

File: nj2k/testData/newJ2k/javaStreamsApi/kt-21467.kt
import java.util.stream.Collectors
import java.util.stream.Stream

internal class Test {
    fun main() {
        val activities: List<String> = Stream.of("12")
                .map { v: String -> v + "nya" }
                .filter { v: String? -> v != null }
                .flatMap { v: String ->
                    Stream.of(v)
                            .flatMap { s: String -> Stream.of(s) }
                }.filter { v: String ->
                    val name: String = v.javaClass.name
                    if (name == "name") {
                        return@filter false
                    }
                    name != "other_name"
                }
                .collect(Collectors.toList())
    }
}